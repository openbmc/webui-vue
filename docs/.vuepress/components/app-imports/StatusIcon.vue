<template>
  <span :class="['status-icon', status]">
    <icon-info v-if="status === 'info'" />
    <icon-success v-else-if="status === 'success'" />
    <icon-warning v-else-if="status === 'warning'" />
    <icon-danger v-else-if="status === 'danger'" />
    <icon-secondary v-else />
  </span>
</template>

<script>
import IconInfo from '@carbon/icons-vue/es/information--filled/20';
import IconCheckmark from '@carbon/icons-vue/es/checkmark--filled/20';
import IconWarning from '@carbon/icons-vue/es/warning--filled/20';
import IconError from '@carbon/icons-vue/es/error--filled/20';
import IconMisuse from '@carbon/icons-vue/es/misuse/20';

export default {
  name: 'StatusIcon',
  components: {
    IconInfo: IconInfo,
    iconSuccess: IconCheckmark,
    iconDanger: IconMisuse,
    iconSecondary: IconError,
    iconWarning: IconWarning
  },
  props: {
    status: {
      type: String,
      default: ''
    }
  }
};
</script>

<style lang="scss">
@import "src/assets/styles/bmc/helpers";
@import "src/assets/styles/bootstrap/helpers";
.status-icon {
  vertical-align: text-bottom;
  &.info {
    color: theme-color('info');
  }
  &.success {
    color: theme-color('success');
  }
  &.danger {
    color: theme-color('danger');
  }
  &.secondary {
    color: gray('600');

    svg {
      transform: rotate(-45deg);
    }
  }
  &.warning {
    color: theme-color('warning');
  }

  svg {
    fill: currentColor;
  }
}
</style>
