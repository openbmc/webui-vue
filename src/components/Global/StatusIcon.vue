<template>
  <span :class="['status-icon', status]">
    <icon-success v-if="status === 'success'" />
    <icon-danger v-else-if="status === 'danger'" />
    <icon-secondary v-else />
  </span>
</template>

<script>
import IconCheckmark from '@carbon/icons-vue/es/checkmark--filled/20';
import IconWarning from '@carbon/icons-vue/es/warning--filled/20';
import IconError from '@carbon/icons-vue/es/error--filled/20';

export default {
  name: 'StatusIcon',
  components: {
    iconSuccess: IconCheckmark,
    iconDanger: IconWarning,
    iconSecondary: IconError
  },
  props: {
    status: {
      type: String,
      default: ''
    }
  }
};
</script>

<style lang="scss" scoped>
.status-icon {
  vertical-align: text-bottom;
  &.success {
    fill: $success;
  }
  &.danger {
    fill: $danger;
  }
  &.secondary {
    fill: $secondary;
  }
}
</style>
