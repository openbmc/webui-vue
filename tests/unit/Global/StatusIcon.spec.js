import { mount, createLocalVue } from '@vue/test-utils';
import StatusIcon from '@/components/Global/StatusIcon';

const localVue = createLocalVue();

describe('StatusIcon.vue', () => {
  const wrapper = mount(StatusIcon, {
    localVue,
    propsData: {
      status: 'info',
    },
  });
  it('should exist', () => {
    expect(wrapper.exists()).toBe(true);
  });
  it('should render icon-info element', () => {
    expect(wrapper.find('.info').exists()).toBe(true);
  });
  it('should render icon-success element', () => {
    const wrapper = mount(StatusIcon, {
      localVue,
      propsData: {
        status: 'success',
      },
    });
    expect(wrapper.find('.success').exists()).toBe(true);
  });
  it('should render icon-warning element', () => {
    const wrapper = mount(StatusIcon, {
      localVue,
      propsData: {
        status: 'warning',
      },
    });
    expect(wrapper.find('.warning').exists()).toBe(true);
  });
  it('should render icon-danger element', () => {
    const wrapper = mount(StatusIcon, {
      localVue,
      propsData: {
        status: 'danger',
      },
    });
    expect(wrapper.find('.danger').exists()).toBe(true);
  });
  it('should render icon-secondary element', () => {
    const wrapper = mount(StatusIcon, {
      localVue,
      propsData: {
        status: 'sec',
      },
    });
    expect(wrapper.find('.status-icon').exists()).toBe(true);
  });
  it('should render correctly', () => {
    expect(wrapper.element).toMatchSnapshot();
  });
});
