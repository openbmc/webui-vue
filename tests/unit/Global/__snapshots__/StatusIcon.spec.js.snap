// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StatusIcon.vue should render correctly 1`] = `
<span
  class="status-icon info"
>
  <svg
    aria-hidden="true"
    fill="currentColor"
    focusable="false"
    height="20"
    preserveAspectRatio="xMidYMid meet"
    viewBox="0 0 32 32"
    width="20"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M16,2A14,14,0,1,0,30,16,14,14,0,0,0,16,2Zm0,5a1.5,1.5,0,1,1-1.5,1.5A1.5,1.5,0,0,1,16,7Zm4,17.125h-8v-2.25h2.875v-6.75h-1.875v-2.25h4.125v9h2.875Z"
    />
    <path
      d="M16,7a1.5,1.5,0,1,1-1.5,1.5A1.5,1.5,0,0,1,16,7Zm4,17.125h-8v-2.25h2.875v-6.75h-1.875v-2.25h4.125v9h2.875Z"
      data-icon-path="inner-path"
      fill="none"
    />
  </svg>
</span>
`;
